bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs.test/src/test/java/org/eclipse/smarthome/config/discovery/usbserial/linux/sysfs/UsbSerialDeviceInformationMatcher.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs;
+
+import java.util.Objects;
+
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDeviceInformation;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+
+/**
+ * Hamcrest matcher for {@link UsbSerialDeviceInformation} objects.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+public class UsbSerialDeviceInformationMatcher extends TypeSafeMatcher<UsbSerialDeviceInformation> {
+
+    private final UsbSerialDeviceInformation other;
+
+    public static UsbSerialDeviceInformationMatcher isUsbSerialDeviceInfo(int vendorId, int productId,
+            String serialNumber, String manufacturer, String product, String serialPort) {
+        return new UsbSerialDeviceInformationMatcher(
+                new UsbSerialDeviceInformation(vendorId, productId, serialNumber, manufacturer, product, serialPort));
+    }
+
+    public static UsbSerialDeviceInformationMatcher isUsbSerialDeviceInfo(UsbSerialDeviceInformation usb) {
+        return isUsbSerialDeviceInfo(usb.getVendorId(), usb.getProductId(), usb.getSerialNumber(),
+                usb.getManufacturer(), usb.getProduct(), usb.getSerialPort());
+    }
+
+    private UsbSerialDeviceInformationMatcher(UsbSerialDeviceInformation other) {
+        this.other = other;
+    }
+
+    @Override
+    protected boolean matchesSafely(UsbSerialDeviceInformation deviceInfo) {
+        if (deviceInfo == null) {
+            return other == null;
+        }
+
+        return Objects.equals(deviceInfo, other)
```

Comment:
The `equals` method for `UsbSerialDeviceInformation` objects only considers serial port and model and vendor ID of the USB device, but not the fields manufacturer, product, and serial. As the checks shall test that all fields are equal, the matcher compares both the objects using `equals`, and the fields that are not included in the equality check.

The interesting question is why the `equals` method is implemented in this way. I did this, because `UsbSerialDeviceInformation` models a serial port provided by a USB device of some type, and I hence included the information about the serial port and about the USB device type. But one might as well include also the other fields into the equality check, and I will do this as it caused confusion here. Thanks for pointing this out.